documentation/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,186 @@
+
+## General Changes
+
+The latest Azure SDK For Go for management modules is using the [Go Modules](https://github.com/golang/go/wiki/Modules) to manage the dependencies. We ship every RP as an individual module to create a more flexible user experience.
+
+The important breaking changes are listed in the following sections:
+
+### Authentication
+
+In old version (`services/**/mgmt/**`), `autorest.Authorizer` is used in authentication process.
+
+In new version (`sdk/**/arm**`), in order to provide a unified authentication based on Azure Identity for all Azure Go SDKs, the authentication mechanism has been re-designed and improved to offer a simpler interface.
+
+To the show the code snippets for the change:
+
+**In old version (`services/**/mgmt/**`)**
+
+```go
+authorizer, err := adal.NewServicePrincipalToken(oAuthToken, "<ClientId>", "<ClientSecret>", endpoint)
+```        
+
+**Equivalent in new version (`sdk/**/arm**`)**
+
+```go
+credential, err = azidentity.NewClientSecretCredential("<TenantId>", "<ClientId>", "<ClientSecret>", nil)
+```
+
+For detailed information on the benefits of using the new authentication classes, please refer to [this page](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/README.md)
+
+### Customized Policy
+
+Because of adopting Azure Core which is a shared library across all Azure SDKs, there is also a minor change regarding how customized policy in configured.
+
+In old version (`services/**/mgmt/**`), we use the `(autorest.Client).Sender`, `(autorest.Client).RequestInspector` and `(autorest.Client).ResponseInspector` properties in `github.com/Azure/go-autorest/autorest` module to provide customized interceptor for the HTTP traffic.
+
+In new version (`sdk/**/arm**`), we use `(armcore.ConnectionOptions).PerCallPolicies` and `(armcore.ConnectionOptions).PerRetryPolicies` in `github.com/Azure/azure-sdk-for-go/sdk/armcore` module instead to inject customized policy to the pipeline.
+
+### Custom HTTP Client
+
+Similar to the customized policy, there are changes regarding how the custom HTTP client is configured as well. You can now use the `(armcore.ConnectionOptions).HTTPClient` option in `github.com/Azure/azure-sdk-for-go/sdk/armcore` module to use your own implementation of HTTP client and plug in what they need into the configuration.
+
+**In old version (`services/**/mgmt/**`)** you cannot customize the HTTP client.
+
+**In new version (`sdk/**/arm**`)**
+
+```go
+httpClient := NewYourOwnHTTPClient{}
+conn := armcore.NewConnection(credential, &armcore.ConnectionOptions{
+    HTTPClient: &httpClient,
+})
+```
+
+### Error Handling
+
+There are some minor changes in the error handling. 
+
+- The errors returned by the SDK now are always of type `runtime.ResponseError` in `github.com/Azure/azure-sdk-for-go/sdk/internal/runtime` package which implements the `HTTPResponse` interface and `NonRetriableError` interface from `github.com/Azure/azure-sdk-for-go/sdk/azcore` package.
```

Comment:
This isn't true, and I don't think calling out `ResponseError` is how we want to direct customers for error handling.

`ResponseError` is a wrapper error that's used to provide access to the underlying `*http.Response`.  Note that we _only_ wrap errors in a `ResponseError` when there's a `*http.Response` to go along with it.  For customers that want access to the raw HTTP response they should use `errors.As()` to obtain access to the `azcore.HTTPResponse` interface.

The `azcore.NonRetriableError` interface is how we identify an error as non-transient.  There are some cases where we aren't adding this (e.g. unmarshalling errors), perhaps we should.

Reply:
I removed this paragraph since it is misleading